contrib/Joshua/scripts/localClusterStart.sh

```diff
@@ -231,12 +236,17 @@ function startFdbServer
 		log 'Failed to display user message'
 		let status="${status} + 1"
 
-	elif ! "${BINDIR}/fdbserver" --knob_disable_posix_kernel_aio=1 -C "${FDBCONF}" -p "${CLUSTERSTRING}" -L "${LOGDIR}" -d "${WORKDIR}/fdb/${$}" &> "${LOGDIR}/fdbserver.log" &
-	then
-		log "Failed to start FDB Server"
-		let status="${status} + 1"
 	else
-		FDBSERVERID="${!}"
+            "${BINDIR}/fdbserver" --knob_disable_posix_kernel_aio=1 -C "${FDBCONF}" -p "${CLUSTERSTRING}" -L "${LOGDIR}" -d "${WORKDIR}/fdb/${$}" &> "${LOGDIR}/fdbserver.log" &
+            fdbpid=$!
+            fdbrc=$?
+            if [ $fdbrc -ne 0 ]
+            then
+                log "Failed to start FDB Server"
+                let status="${status} + 1"
+            else
+                FDBSERVERID="${fdbpid}"
+            fi
```

Comment:
Why was this changed from the previous code? It seems to try to do the same as before with more code.

Reply:
This one was really tricky.  The original code was returning a wrong return value for us.  It can be environment dependent undefined behavior or bug in bash.
It's similar to this test case:
```
if ! true &> true.out &
then
    echo "failed"
else
    echo "succeeded"
fi
```